src/chrome/cdtpDebuggee/eventsProviders/cdtpOnScriptParsedEventProvider.ts

```diff
@@ -0,0 +1,257 @@
+
+        await this.registerRuntimeAndDevelopmentSourcesRelationships(script);
+
+        return script;
+    }
+
+    private sourceMap(): Promise<SourceMap> {
+        return this._sourceMapTransformer.scriptParsed(this.runtimeSourcePath.canonicalized, this._scriptParsedEvent.sourceMapURL);
+    }
+
+    protected abstract createScript(executionContext: IExecutionContext, sourceMapperProvider: (script: IScript) => IMappedSourcesMapper,
+        mappedSourcesProvider: (script: IScript) => IdentifiedLoadedSource<string>[]): Promise<IScript>;
+
+    protected abstract registerRuntimeAndDevelopmentSourcesRelationships(script: IScript): Promise<void>;
+
+    private mappedSources(sourceMapper: IMappedSourcesMapper): IdentifiedLoadedSource[] {
+        return sourceMapper.sources.map((path: string) => this.obtainLoadedSource(parseResourceIdentifier(path), SourceScriptRelationship.Unknown));
+    }
+
+    private sourceMapper(script: IScript, sourceMap: SourceMap): IMappedSourcesMapper {
+        const sourceMapper = sourceMap
+            ? new MappedSourcesMapper(script, sourceMap)
+            : new NoMappedSourcesMapper(script);
+        return sourceMapper;
+    }
+
+    protected scriptRange(runtimeSource: ILoadedSource<CDTPScriptUrl>) {
+        const startPosition = new Position(createLineNumber(this._scriptParsedEvent.startLine), createColumnNumber(this._scriptParsedEvent.startColumn));
+        const endPosition = new Position(createLineNumber(this._scriptParsedEvent.endLine), createColumnNumber(this._scriptParsedEvent.endColumn));
+        const scriptRange = new RangeInResource(runtimeSource, startPosition, endPosition);
+        return scriptRange;
+    }
+
+    protected obtainLoadedSource(sourceUrl: IResourceIdentifier, sourceScriptRelationship: SourceScriptRelationship): IdentifiedLoadedSource {
+        return this._loadedSourcesRegistry.getOrAdd(sourceUrl, provider => {
+            return IdentifiedLoadedSource.create(sourceUrl, sourceScriptRelationship, provider);
+        });
+    }
+}
+
+class IdentifiedScriptCreator extends ScriptCreator {
+    private readonly runtimeSource = _.memoize(() => this.obtainRuntimeSource());
+    private readonly developmentSource = _.memoize(() => this.obtainDevelopmentSource());
+
+    protected async createScript(executionContext: IExecutionContext, sourceMapperProvider: (script: IScript) => IMappedSourcesMapper,
+        mappedSourcesProvider: (script: IScript) => IdentifiedLoadedSource<string>[]): Promise<IScript> {
+        return Script.create(executionContext, this.runtimeSource(), await this.developmentSource(), sourceMapperProvider, mappedSourcesProvider, this.scriptRange(this.runtimeSource()));
+    }
+
+    private obtainRuntimeSource(): IdentifiedLoadedSource<CDTPScriptUrl> {
+        // This is an heuristic. I think that if the script starts on (0, 0) then that means the file is a script file, and not an .html file or something which is a script and something else
```

Comment:
Why do we need to know this?

Reply:
It's to show the loaded script/dynamic documents. If the dynamic document is a JavaScript script, we can use Debugger.getScriptSource to get the contents of the .js file and show it to the user.
If the scripts are inline scripts in an .html file, then we want to show the full source of the .html file, including the HTML code. To do that we use the Page.getResourceContent API instead.
We use this field to figure if this is a script.js vs an index.html file, and use the proper API.